src/Arcus.WebApi.OpenApi.Extensions/OAuthAuthorizeOperationFilter.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Authorization;
+using Swashbuckle.AspNetCore.Swagger;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Arcus.WebApi.OpenApi.Extensions
+{
+    /// <summary>
+    /// A Swashbuckle operation filter that adds OAuth security definitions to authorized API operations.
+    /// </summary>
+    public class OAuthAuthorizeOperationFilter : IOperationFilter
+    {
+        private readonly IEnumerable<string> _scopes;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OAuthAuthorizeOperationFilter"/> class.
+        /// </summary>
+        public OAuthAuthorizeOperationFilter(IEnumerable<string> scopes)
+        {
+            _scopes = scopes;
+        }
+
+        public void Apply(Operation operation, OperationFilterContext context)
```

Comment:
I would also verify the incoming parameters.
`Guard` them both and each property that you're using afterwards for `null`.

Reply:
I could do that, but I wonder if it adds any value here.   This operation is being called by Swashbuckle and we do not have any control over this.  I guess we should rely on the stability of Swashbuckle that the operation gets called with the correct parameters in this case ?

Reply:
I tend to agree, if it gives null it's a Swashbuckle issue which we cannot fix anyway

Reply:
True on that we could not fix it, but at least we handle it.
Was an idea, not blocking. Maybe I'm sometimes a bit to paranoid.

Reply:
I think we're good to go here, resolving!